core/play/src/main/scala/play/api/libs/Files.scala

```
@@ -256,11 +256,13 @@ object Files {
     // Keeping references ensures that the FinalizablePhantomReference itself is not garbage-collected.
     private val references = Sets.newConcurrentHashSet[Reference[TemporaryFile]]()
 
-    private val TempDirectoryPrefix = "playtemp"
+    private val TempDirectoryPrefix        = "playtemp"
+    private var TempFolderCreated: Boolean = false
```

Hi, @gaeljw! Thanks a lot to your contribution 👍 
In common I agree with this approach, but personally, I would prefer to change type of private `playTempFolder` and use `Option/Either` for them instead of add a new boolean field. It'll be a more idiomatic way instead of manually sync a state of two `Path` and `Boolean` vars 😉

---

@ihostage I agree it's not the best pattern but I wanted to keep the laziness of `playTempFolder`.

Are you suggesting to have a mutable `Option[Path]` and replace the `lazy val` creation of the folder by something like `playTempFolder.getOrElse(create...)` in the `create` method?

---

No, I thought like this
```scala
private lazy val playTempFolder: Either[Throwable, Path] =
      try {
        ...
        Right(tmpFolder)
      } catch {
        case t: Throwable => Left(t)
      }
```
But now, I realize that this approach still creates and deletes a temporary folder unnecessarily, just potentially hide an error message in the log via the stop hook implementation. I don't like this. 😄 
Need to think 🤔

---

I implemented the alternative you talked about in #13110

What do you guys think / prefer?
I don't really care.

---

Slight preference for my initial proposal that I find more explicit but I'm fine with yours as well.

Let's get one of the two merged :)

---

The alternative suggestion does not work anyway, because it can cause race conditions, see https://github.com/playframework/playframework/pull/13110#discussion_r1957281123

Therefore lets just merge this PR finally.